drivers/ethernet/eth_native_posix.c

```diff
@@ -48,35 +48,6 @@
 #define ETH_HDR_LEN sizeof(struct net_eth_hdr)
 #endif
 
-#if defined(CONFIG_NET_LLDP)
```

Comment:
I do not think this is a good thing. Now caller cannot specify lldpdu that is to be sent. The LLDP API should not really care what data is being sent by the ethernet driver. Just drop this commit from this set. Other commits looked ok.

Reply:
I was thinking how to avoid code duplication, we have the same code for potentially all ethernet (and slip) drivers. Maybe we could move it to ethernet_init() @jukkar ?

Reply:
@jukkar I dropped last commit from this PR, I have other commits implementing optional TLVs to be sent after this PR.